### ISSUE TITLE
disallow casting temporal to integer

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#121]
in pandas this does something really strange

```python
In [3]: pd.to_datetime(['2020', None]).astype('int64')
Out[3]: Index([1577836800000000000, -9223372036854775808], dtype='int64')
```

Users shouldn't be relying on phyisical representations anyway, we should nudge them towards `.dt.microseconds` or whatever it is they're looking for

**Comment by FBruzzesi** [comment]
Polars seems to allow it, except null is preserved:

```py
from datetime import datetime
import polars as pl

pl.Series([
    datetime(year=2020, month=1, day=1, hour=0, second=0, minute=0),
    None
]).cast(pl.Int64)
```

```terminal
shape: (2,)
Series: '' [i64]
[
        1577836800000000
        null
]
```

**Comment by MarcoGorelli** [comment]
yeah I think so, I think it'd be better to nudge people towards `.dt.timestamp` / `.dt.microsecond` / etc. depending on which they want, it should be more robust than casting

**Comment by DeaMariaLeon** [comment]
Do we still want this?

**Comment by MarcoGorelli** [comment]
yes I think we should raise where possible when trying to cast a datetime to a numeric type. At least for pandas-like and PyArrow (those are the ones people are most likely to test, so if they find that it raises in those cases, they'll be less likely to use it)
